src/main/java/mate/academy/bookdao/BookDaoImpl.java

```diff
@@ -0,0 +1,141 @@
+        String sql = "SELECT COUNT(*) AS total_rows FROM book";
+        try (Connection connection = ConnectionUtil.getConnection();
+                        PreparedStatement statement = connection.prepareStatement(sql);
+                        ResultSet resultSet = statement.executeQuery()) {
+            if (resultSet.next()) {
+                return resultSet.getInt("total_rows");
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't get row count", e);
+        }
+        return 0;
+    }
+
+    @Override
+    public boolean clear() {
+        String sql = "TRUNCATE TABLE book";
+        try (Connection connection = ConnectionUtil.getConnection();
+                        Statement statement = connection.createStatement()) {
+            int affectedRows = statement.executeUpdate(sql);
+            return affectedRows >= getRowsCount();
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't clear table", e);
+        }
+    }
+
+    @Override
+    public Book create(Book book) {
+        String sql = "INSERT INTO book(title, price) VALUES(?, ?)";
+        try (Connection connection = ConnectionUtil.getConnection();
+                        PreparedStatement statement = connection
+                                .prepareStatement(sql, Statement.RETURN_GENERATED_KEYS)) {
+            statement.setObject(1, book.getTitle());
+            statement.setInt(2, book.getPrice());
+            int affectedRows = statement.executeUpdate();
+            if (affectedRows < 1) {
+                throw new RuntimeException("Book could not be created");
+            }
+            ResultSet generatedKeys = statement.getGeneratedKeys();
+            if (generatedKeys.next()) {
+                long id = generatedKeys.getLong(1);
+                book.setId(id);
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't add a new book " + book.getTitle(), e);
+        }
+        return book;
+    }
+
+    @Override
+    public Optional<Book> findById(Long id) {
+        String sql = "SELECT * FROM book WHERE id = ?";
+        try (Connection connection = ConnectionUtil.getConnection();
+                        PreparedStatement preparedStatement = connection.prepareStatement(sql)) {
+            preparedStatement.setLong(1, id);
+            ResultSet resultSet = preparedStatement.executeQuery();
+            if (resultSet.next()) {
+                String title = resultSet.getString("title");
+                int price = resultSet.getInt("price");
+                Book book = Book.of(id, title, price);
+                return Optional.of(book);
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't find book by id " + id, e);
+        }
+        return Optional.empty();
+    }
+
+    @Override
+    public List<Book> findAll() {
+        String sql = "SELECT * FROM book";
```

Comment:
The table name should be `books` instead of `book` according to the SQL style guide in the checklist. Additionally, use PreparedStatement over Statement, even for a static query with no parameters in `findAll()` method.